daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/dsom/SDE.scala

```diff
@@ -120,20 +120,14 @@ class SchemaDefinitionWarning(
     args: _*,
   ) {
 
-  def this(sc: SchemaFileLocation, kind: String, args: Any*) =
-    this(None, Some(sc), None, kind, args: _*)
+  def this(w: WarnID, sc: SchemaFileLocation, kind: String, args: Any*) =
+    this(w, Some(sc), None, kind, args: _*)
 
   override def isError = false
   def modeName = "Schema Definition"
 
-  override def toString = warnID match {
-    case Some(id) =>
-      super.toString + "\n" +
-        s"To suppress this warning, add '${id}' to the " +
-        "daf:suppressSchemaDefinitionWarnings element of a Daffodil defineConfig " +
-        "used when processing"
-    case None => super.toString
-  }
+  override def getMessage = super.getMessage + s" (id: ${warnID})"
```

Comment:
I believe this puts the warn id after file/line number info if it's included in the SDW (which is almost always is), which might be confusing? Does the warn id want to be closer to the actual message?

Maybe we do somehing like:
```scala
class SchemaDefinitionWarning(
  ...
) extends SchemaDefinitionDiagnosticBase(
    ...,
    Maybe(kind + s" (id: ${warnID})")
    ...,
) {
```

So we actually append the warnID to the diagnostic format string so it appears before any location information.

Reply:
I can look into that. I tried to implement the first idea where the ID was right after the SDW text but the information needed seemed to be private.

Reply:
Ok, I updated it to that which doesn't require opening private definitions.